libarchive/write.py

```diff
@@ -73,6 +73,47 @@ def add_files(self, *paths):
                         write_finish_entry(write_p)
                         entry_clear(entry_p)
 
+    def add_entry_from_memory(self, entry_path, entry_size, entry_data):
```

Comment:
@Changaco: changed- I removed these private methods and moved code to `add_file_from_memory`